routes/course.js

```diff
@@ -223,37 +246,53 @@ router.patch("/api/courses/changeDistribution/:course_id"),
   };
 */
 
-//delete a course given course id
-//update associated distribution credits
+// delete a course given course id
+// update associated distribution credits
 router.delete("/api/courses/:course_id", (req, res) => {
-  const c_id = req.params.course_id;
+  const c_id = ObjectId(req.params.course_id);
   courses
     .findByIdAndDelete(c_id)
-    .then((course) => {
-      course.distribution_ids.forEach((id) => {
-        distributions
-          .findByIdAndUpdate(
-            id,
-            { $pull: { courses: c_id } },
-            { new: true, runValidators: true }
-          )
-          .then((distribution) =>
-            distributionCreditUpdate(distribution, course, false)
-          )
-          .catch((err) => errorHandler(res, 500, err));
-      });
+    .then(async (course) => {
+      let updatedDists = [];
+      let updatedFines = [];
+      // remove course from distributions
+      for (let id of course.distribution_ids) {
+        await distributions
+          .findById(id)
+          .then(async (distribution) => {
+            await requirementCreditUpdate(distribution, course, false);
+            // remove course from fineReqs
```

Comment:
are you just adjusting the fineReq credits? there is no actual course id being removed from fineReq right

Reply:
right we're just adjusting credits because fineReq objects do not maintain a `course_ids` array